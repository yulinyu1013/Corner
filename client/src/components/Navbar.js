import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { getUnreadNotificationNum } from '../fetchers/notificationApis';
import logo from '../images/logo.png';
// import userPic from '../images/logo.png';
import notification from '../images/notification.png';
import '../styles/navbar.css';

function Navbar({ currentUser, isLoggedIn }) {
  const [notificationNum, setNotificationNum] = useState(0);
  console.log(currentUser);
  // const loggedIn = isLoggedIn;
  const history = useHistory();
  //   const [loggedIn, setLoggedIn] = useState(true);
  const redirectToNotification = () => {
    history.push(`/notification/${currentUser.id}`);
  };

  const user = currentUser === {} ? null : currentUser;

  useEffect(async () => {
    console.log(isLoggedIn);
    if (isLoggedIn) {
      getUnreadNotificationNum(currentUser.id).then((data) => {
        setNotificationNum(data);
      });
    }
  }, [isLoggedIn, currentUser]);

  const redirectToHomepage = () => {
    history.push('/');
  };
  return (
    <div className="navgation-bar">
      <div className="logo" role="presentation" onClick={() => redirectToHomepage()}>
        <img className="logo" src={logo} alt="logo" />
      </div>
      {isLoggedIn && (
      <div className="user-info">
        <img className="small-icon" src={notification} alt="notification" role="presentation" onClick={() => redirectToNotification()} />
        <p>{`(${notificationNum})`}</p>
        <img className="user-pic" src={!user || !user.avatar ? logo : user.avatar} alt="user-avatar" />
        <p className="user-name">{user && user.name}</p>
      </div>
      )}
    </div>
  );
}

export default Navbar;
