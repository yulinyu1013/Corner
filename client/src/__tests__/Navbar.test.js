import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import user from '../images/user.png';

const currentUser = {
  id: '1',
  name: 'test',
  avatar: {user},
}

it('renders Navbar correctly when user not loggedin', () => {
  const tree = renderer.create(<Navbar isLoggedIn={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Navbar correctly when user not loggedin', () => {
  const tree = renderer.create(<Navbar isLoggedIn={true} currentUser={currentUser} />).toJSON();
  expect(tree).toMatchSnapshot();
});
