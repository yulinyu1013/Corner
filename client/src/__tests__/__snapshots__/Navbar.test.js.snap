// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly when user not loggedin 1`] = `
<div
  className="navgation-bar"
>
  <div
    className="logo"
    onClick={[Function]}
    role="presentation"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </div>
</div>
`;

exports[`renders Navbar correctly when user not loggedin 2`] = `
<div
  className="navgation-bar"
>
  <div
    className="logo"
    onClick={[Function]}
    role="presentation"
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </div>
  <div
    className="user-info"
  >
    <img
      alt="notification"
      className="small-icon"
      onClick={[Function]}
      role="presentation"
      src="notification.png"
    />
    <p>
      (0)
    </p>
    <img
      alt="user-avatar"
      className="user-pic"
      src={
        Object {
          "user": "user.png",
        }
      }
    />
    <p
      className="user-name"
    >
      test
    </p>
  </div>
</div>
`;
